src/ophyd_async/core/detector.py

```diff
@@ -51,20 +51,25 @@ class DetectorTrigger(str, Enum):
     variable_gate = "variable_gate"
 
 
-@dataclass(frozen=True)
-class TriggerInfo:
+class TriggerInfo(BaseModel):
     """Minimal set of information required to setup triggering on a detector"""
 
     #: Number of triggers that will be sent
-    num: int
+    number: int = Field(gt=0)
```

Comment:
We should document here that `==0` means go forever as this is the convention areaDetector chose.
```suggestion
    #: Number of triggers that will be sent, 0 means infinite
    number: int = Field(ge=0)
```